### DISCUSSION TITLE
Does code app run on a VM

### DISCUSSION THREAD

**Discussion by hg0428** [thebaselab/codeapp#506] [Q&A]
Does code app run on a VM

**Comment by UntrustedTech** [comment]
Are you asking if the code app runs on a virtual machine? What made you think of this conclusion? Just curious... And if so the answer is no its not running on a vm.

**Comment by hg0428** [comment]
I was mainly just curious.
But, I mentioned it in a programming community and everyone was saying "The only way to run code on mobile is with a VM"
I was quite sure it didn't run on a VM, but I just asked here real quick to be sure.

**Reply by UntrustedTech** [reply]
I mean in a way they aren't wrong that when it comes to coding mobile apps at least in the android side there is a lot of insulation involved and same thing with iOS, I'm still trying to understand how this app functions but for the most part at least when compiling and running things it does involve some local em, but for the rest, its all just the app.

**Reply by UntrustedTech** [reply]
Don't quote me i may be wrong

**Reply by bummoblizard** [reply]
No. iOS apps are sandboxed but do not run in a VM. We use interpreters / web assembly to run binaries to workaround the code signing requirement.

https://support.apple.com/en-gb/guide/security/sec15bfe098e/web

**Reply by UntrustedTech** [reply]
Ohhhhhh ok I see now, im still learning about swift and how it's worked around its limitations.

**Reply by hg0428** [reply]
> No. iOS apps are sandboxed but do not run in a VM. We use interpreters / web assembly to run binaries to workaround the code signing requirement.
> 
> https://support.apple.com/en-gb/guide/security/sec15bfe098e/web

Yes, thats what I tried to tell them to begin with.
Thanks for confirming.
